src/Gizmos/positionGizmo.ts

```diff
@@ -192,6 +201,81 @@ export class PositionGizmo extends Gizmo {
         return this._scaleRatio;
     }
 
+    /**
+     * Builds Gizmo Axis Cache to enable features such as hover state preservation and graying out other axis during manipulation
+     * @param mesh Axis gizmo mesh
+      @param cache display gizmo axis thickness
+     */
+    public addToAxisCache(mesh: Mesh, cache: any) {
+        this.gizmoAxisCache.set(mesh, cache);
+    }
+
+    /**
+     * Subscribes to pointer up, down, and hover events. Used for responsive gizmos.
+     */
+    public subscribeToPointerObserver(): void {
+        const pointerObserver = this.gizmoLayer.utilityLayerScene.onPointerObservable.add((pointerInfo) => {
+            if (pointerInfo.pickInfo) {
+                // On Hover Logic
+                if (pointerInfo.type === PointerEventTypes.POINTERMOVE) {
+                    if (this.dragging) { return; }
+                    this.gizmoAxisCache.forEach((statusMap, parentMesh) => {
+                        const isHovered = pointerInfo.pickInfo && (parentMesh.getChildMeshes().indexOf((pointerInfo.pickInfo.pickedMesh as Mesh)) != -1);
+                        const material = isHovered || statusMap.active ? statusMap.hoverMaterial : statusMap.material;
+                        parentMesh.getChildMeshes().forEach((m) => {
+                            if (m.name !== 'ignore') {

```

Comment:
is there a better way than comparing strings to check if this mesh needs to be ignored?
If you can't find any, please set this string to be a static member of one of the main classes, and reuse the variable, instead of having `"ignore"`  everywhere.

Reply:
I agree with Raanan. Can you build up a list of ignored meshes. Or separate hierarchy of ignore mesh under 2 different parents.